scapy/contrib/isotp.py

```diff
@@ -168,6 +172,110 @@ def defragment(can_frames, use_extended_addressing=None):
         return results[0]
 
 
+class ISOTPHeader(CAN):
+    name = 'ISOTPHeader'
+    fields_desc = [
+        FlagsField('flags', 0, 3, ['error',
+                                   'remote_transmission_request',
+                                   'extended']),
+        XBitField('identifier', 0, 29),
+        ByteField('length', None),
+        ThreeBytesField('reserved', 0),
+    ]
+
+    def extract_padding(self, p):
+        return p, None
+
+    def post_build(self, p, pay):
+        """
+        This will set the ByteField 'length' to the correct value.
+        """
+        if self.length is None:
+            p = p[:4] + chb(len(pay)) + p[5:]
+        return p + pay
+
+    def guess_payload_class(self, payload):
+        """
+        ISOTP encodes the frame type in the first nibble of a frame.
+        """
+        if six.PY2:
+            t = (ord(payload[0]) & 0xf0) >> 4
+        else:
+            t = (int(payload[0]) & 0xf0) >> 4
```

Comment:
`orb` could be used here

Reply:
Totally. This would allow you to remove this `six.PY2` test.

Reply:
fixed

Reply:
thanks for this remark